### DISCUSSION TITLE
Attempt to simplify the serve command for prod

### DISCUSSION THREAD

**Discussion by ganigeorgiev** [pocketbase/pocketbase#3190] [Announcements]
**TL;DR:** With the next release you'll be able to run on prod `./pocketbase serve example.com` and this should start a HTTPS server out of the box without requiring the users to edit their `/etc/hosts` file when a direct mapping between the domain and the vm local network interface is missing.

---

A lot of users reported issues related to the auto issued Let's encrypt certificate when using the PocketBase executable with a domain name (see https://github.com/pocketbase/pocketbase/pull/2795, https://github.com/pocketbase/pocketbase/discussions/3179, https://github.com/pocketbase/pocketbase/discussions/1510#discussioncomment-6807838, https://github.com/pocketbase/pocketbase/issues/107#issuecomment-1331714755, etc.)

The issues originate from the fact that when you have something like:

```
./pocketbase serve --http=example.com:80 --https=example.com:443
```

we currently try to bind **only** to the single IPv4/IPv6 address that `example.com` is resolving to.

This is not a problem when the domain is managed within the VPS vendor interface (eg. in [Hetzner DNS Console](https://www.hetzner.com/dns-console), [DigitalOcean Domains control panel](https://docs.digitalocean.com/products/networking/dns/), etc.), since this "mapping" between the VPS network interface and the domain is handled automatically by the vendor (_most of the times_).

But in some cases, users usually have to register manually an entry in their `/etc/hosts` file that describes the local network interface - domain relation (_see https://github.com/pocketbase/pocketbase/issues/107#issuecomment-1331714755_), which may no be always immediately clear.

This could be avoided if we listen to all local IPv4/IPv6 interfaces, aka. `0.0.0.0`/`[::]` (_most reverse proxies do the same thing_). This is usually recommended/"safe" only for services that we want to be public accessible (_as it is in our case when a Let's encrypt certificate is issued for a public domain_). But it could also has some security implications, specifically on local dev environments with misconfigured/disabled firewall, because users may accidentally and unknowingly expose their service to the public (this is also why the default PocketBase address is `127.0.0.1:8090` and not `0.0.0.0:8090`), so the default listener addresses ideally should be conditional.

With that said, in order to have a smoother out of the box deployment experience, with the next release the `serve` command will support domain(s) names as optional argument(s) that will auto set the default values behind the scene of the related `--http` and `--https` listener addresses. When at least one domain name argument is specified, the default flags values will be autoset to `--http="0.0.0.0:80"` and `--https="0.0.0.0:443"`, otherwise - the current "safe" local dev defaults will be used (aka. `--http="127.0.0.1:8090"` and `--https=""`)

For example:

```sh
./pocketbase serve example.com sub.example.com
```

**Note that the `--http` and `--https` flags are still used and not deprecated!**
They allow explicitly specifying the listener addresses and users can use them in combination with the above for example to listen to all IPv6 interfaces, instead of the default IPv4:

```sh
./pocketbase serve example.com --http="[::]:80" --https="[::]:443"
```

Additionally, in case the domain argument is missing, the existing behavior is preserved, aka.:

```sh
./pocketbase serve --http="example.com:80" --https="example.com:443"
```

would be more-or-less the same as running:

```sh
./pocketbase serve example.com --http="93.184.216.34:80" --https="93.184.216.34:443"
```
_(`93.184.216.34` is the IP of the `example.com` DNS A record)_

**Comment by mpiorowski** [comment]
Great idea. As an additional solution, You can also place an nginx proxy in front of the http pocketbase , that will take care of ssl, caching, etc. Ofc it's more work, but You also gain more control over the flow :)

**Comment by moofone** [comment]


**Reply by ganigeorgiev** [reply]
_Just to prevent eventual confusion in case someone stumble here - the above post is reply to https://github.com/pocketbase/pocketbase/discussions/4952._
